distributed/diagnostics/progress.py

```diff
@@ -358,20 +355,16 @@ def insert(self, timestamp, name, group):
                 count = states["erred"] + states["memory"] + states["released"]
 
                 if count - pcount >= self._deltas[name]:
-                    state = {
-                        "memory": group.states["memory"],
-                        "erred": group.states["erred"],
-                        "waiting": group.states["waiting"],
-                        "released": group.states["released"],
-                        "processing": group.states["processing"],
-                    }
-                    self.states[name].append((timestamp, state))
+                    self.states[name].append((timestamp, group.states.copy()))
+                    self.all_durations[name].append(
+                        (timestamp, group.all_durations.copy())
+                    )
                     self.nbytes[name].append((timestamp, group.nbytes_total))
 
     def transition(self, key, start, finish, *args, **kwargs):
```

Comment:
I wonder if here we can use the scheduler `transition_counter` (see https://github.com/dask/distributed/pull/4934/files) or maybe the `transition_table` (https://github.com/dask/distributed/blob/e5eb40c597dba0358221500fda49d653b7cf1f3a/distributed/scheduler.py#L1928) although it might be an overkill.